code/modules/mob/mob.dm

```diff
@@ -106,6 +106,11 @@
 // blind_message (optional) is what blind people will hear e.g. "You hear something!"
 
 /mob/visible_message(message, self_message, blind_message)
+	if(!isturf(src.loc)) // cant see what do mob do in a locker or something
+		if(self_message)
+			var/msg = self_message
```

Comment:
there's no point to defining `msg` here if you use it once and are assigning an already available parameter `self_message` without modifying it.

Reply:
changed it in update